src/App.js

```diff
@@ -1,22 +1,229 @@
+                                  })}
+                                >
+                                  {item.slug}
+                                </div>
+                                {dragProvided.placeholder}
+                              </div>
+                            )}
+                          </Draggable>
+                        ))
+                      }
+                    </div>
+                    {dropProvided.placeholder}
+                  </div>
+                )}
+                </Droppable>
+                )}/>
+
+          </DragDropContext>
+          <button onClick={() => this.setState({isOpen: !this.state.isOpen})}>open</button>
+          <hr/>
+
+          <Downshift
+            itemToString={itemToString}
+            isOpen={this.state.isOpen}
+            selectedItem={this.state.selectedLessons}
+            onStateChange={this.handleStateChange}
+          >
+            {({
+                getInputProps,
+                getItemProps,
+                isOpen,
+              }) => (
+              <div>
+                {!isOpen
+                  ? null
+                  : <div>
+                    <input {...getInputProps()} />
+                    {
+                      this.state.allLessons.map((item) => (
+                        <div key={item.slug} {...getItemProps({
+                          onClick: () => this.selectItem(item),
+                          item,
+                          style: {fontWeight: some(this.state.selectedLessons, item) ? 'bold' : 'normal'}
+                        })}>
+                          <input readOnly={true} type="checkbox" checked={some(this.state.selectedLessons, item)}/>{item.slug}
+                        </div>
+                      ))
+
+                    }
+                  </div>}
+              </div>
+            )}
+          </Downshift>
         </div>
       </StaticRouter>
     )
   }
 }
 
-export default App
\ No newline at end of file
+export default App
+
+function noop() {
+}
+
+const reorder = (list, startIndex, endIndex) => {
+  const result = Array.from(list);
+  const [removed] = result.splice(startIndex, 1);
+  result.splice(endIndex, 0, removed);
+
+  return result;
+}
+
+function unwrapArray(arg, defaultValue) {
+  arg = Array.isArray(arg) ? /* istanbul ignore next (preact) */ arg[0] : arg
+  if (!arg && defaultValue) {
+    return defaultValue
+  } else {
+    return arg
+  }
+}
+
+function composeEventHandlers(...fns) {
+  return (event, ...args) =>
+    fns.some(fn => {
+      fn && fn(event, ...args)
+      return event.defaultPrevented
+    })
+}
+
+function itemToString(item) {
+  return item.slug
+}
+
+class List extends React.Component {
+
+  getStateAndHelpers = () => {
+    const {getItemProps, selectItem} = this
+
+    return {
+      getItemProps,
+      selectItem
+    }
+  }
+
+  selectItem = () => {
+    console.log('do select')
+  }
+
+  getItemProps = ({onClick, ...rest}) => {
```

Comment:
not a big fan of this method. IMO it makes it harder to understand. 
I'll look into this to find an alternate solution that doesn't hijack the code and provides the same functionality.

Reply:
this approach allows me the user to pass in arbitrary properties. So our known/required properties are captured and utilized, but the consumer has the ultimate flexibility to affect change on the item element as required for their unknown (to us), specific use case.

it's a pattern for highly customizable reusable components and isn't as difficult to understand once it is recognized.

You could elaborate on "hijack the code". Always interesting to hear the other side of this specific debate.